File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int curr = 1;
        int prev = 0;

        while(curr < nums.size()){
            if(nums[curr] == 0 && nums[prev] == 0){
                curr++;
                continue;
            }
            else if(nums[prev] == 0 && nums[curr] != 0){
                swap(nums[prev], nums[curr]);
                curr++;
                prev++;
            }else{
                prev++;
                curr++;
            }
        }
    }
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int st = 0;
        int last = 1;

        while(last<nums.length){
            if(nums[last] == 0 && nums[st] != 0){
                last++;
                st++;
            }else if(nums[last] == 0){
                last++;
            }
            else if(nums[st] == 0 && nums[last] != 0){
                int temp = nums[st];
                nums[st] = nums[last];
                nums[last] = temp;
                last++;
                st++;
            }else{
                st++;
                last++;
            }
        }
    }
}